### ISSUE TITLE
no-node-access rule output false-positive warning for FireEvent after 7.5.4

### ISSUE THREAD

**Issue by pstachula-dev** [testing-library/eslint-plugin-testing-library#1038]
### Have you read the Troubleshooting section?

Yes

### Plugin version

7.5.4

### ESLint version

9.30.1

### Node.js version

22.14

### Bug description

False positive error for fireEvent from:
`import { fireEvent } from '@testing-library/react';`

### Steps to reproduce

1. Bump eslint-plugin-testing-library from 7.5.3 to 7.5.4

### Error output/screenshots

<img width="933" height="184" alt="Image" src="https://github.com/user-attachments/assets/e0cb3023-15ea-417b-ac0b-026929cb1ec5" />

### ESLint configuration

```
  {
    files: ['**/*.test.{ts,tsx}'],
    ...pluginTestingLibrary.configs['flat/react'],
  },
```

### Rule(s) affected

no-node-access

### Do you want to submit a pull request to fix this bug?

No

**Comment by y-hsgw** [comment]
I tried reproducing it locally using the same version, but I couldn't reproduce the issue 😢
Also, as shown in the following test case, I believe that usage of `fireEvent` should already be supported:
https://github.com/testing-library/eslint-plugin-testing-library/blob/7812dae547fd2237f90d9db37dad8c11a0dbee89/tests/lib/rules/no-node-access.test.ts#L214-L221

Would it be possible for you to share a sample code snippet or a link to the relevant repository instead of a screenshot?

**Comment by pstachula-dev** [comment]
Ok, so  I will create reproduction repo

**Comment by pstachula-dev** [comment]
@y-hsgw I found the issue - very wierd.

https://stackblitz.com/edit/vitejs-vite-wqratcpq?file=src%2Fexample.test.tsx,eslint.config.mjs,package.json,src%2Fcustom-render.tsx&terminal=dev

tldr:
- when I use re-export pattern with custom-render, then I have errors

```ts
import { fireEvent, screen } from './custom-render';
// import { fireEvent, screen } from '@testing-library/react';

it('should fetch users with Authorization header', async () => {
  fireEvent.click(screen.getByTestId('fetch-users-btn'));
});

```

**Comment by Belco90** [comment]
@pstachula-dev thanks for reporting. It seems the new Testing Library resolver is colliding already with the Aggressive Reporting.

@y-hsgw I think if the util is not considered as a Testing Library resolver, we should apply some Aggressive Reporting assumptions with the default Testing Library util names. What do you think?

**Comment by dleung107** [comment]
I'm seeing a similar issue with `userEvent` but only if the `user` object is created in the before each.

Eslint Error:
```js
import {screen} from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('Testing', () => {
  let user;

  beforeEach(() => {
    user = userEvent.setup();
  });

  it('test 1', async () => {
    await user.click(screen.getByRole('button'));
  });
});
```

No Eslint Error:
```js
import {screen} from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('Testing', () => {
  it('test 1', async () => {
    const user = userEvent.setup();
    await user.click(screen.getByRole('button'));
  });
});
```

**Comment by rdamazio** [comment]
I'm seeing exactly what @dleung107 reported too.

Example CI failure if you want something concrete: https://github.com/rdamazio/efis-editor/actions/runs/16232609553/job/45838131796?pr=247

**Comment by y-hsgw** [comment]
@Belco90 
I might not have fully understood the comment, but I believe the main issue here is that `resolveToTestingLibraryFn` doesn't currently support Aggressive Reporting properly.
For example, cases with relative imports aren’t handled correctly 😢
(Sorry, at the time of that [comment](https://github.com/testing-library/eslint-plugin-testing-library/pull/1033#discussion_r2194959485), I was mistakenly thinking that relative paths weren’t supported.)

I think we can fix this by adjusting the logic to follow what's done in `detect-testing-library-utils`, particularly around this part:

https://github.com/testing-library/eslint-plugin-testing-library/blob/7812dae547fd2237f90d9db37dad8c11a0dbee89/lib/create-testing-library-rule/detect-testing-library-utils.ts#L963-L966

**Comment by AlexKrupko** [comment]
The same issue after upgrading to 7.5.4. It warns on `userEvert.click` returned from the custom `render` function 
```ts
test('....', async () => {
    const {userEvent} = render(....);
    // ..... 
    await userEvent.click(element);
});
```

**Comment by dennisoelkers** [comment]
I am seeing something similar, when `fireEvent` was imported from a wrapper module:

```
import { render, screen, fireEvent } from 'wrappedTestingLibrary';

[...]
    fireEvent.click(closeBtn);
```

raises `testing-library/no-node-access`.

**Comment by pstachula-dev** [comment]
@dennisoelkers Yes Dennis, this core issue :) https://github.com/testing-library/eslint-plugin-testing-library/issues/1038#issuecomment-3058152180

**Comment by zyhou** [comment]
> The same issue after upgrading to 7.5.4. It warns on `userEvert.click` returned from the custom `render` function
> 
> test('....', async () => {
>     const {userEvent} = render(....);
>     // ..... 
>     await userEvent.click(element);
> });

the PR doesn't fix that case

**Comment by rdamazio** [comment]
I can confirm that this is not fixed in 7.6.3, the examples I linked to above are still falsely triggering (151 false positives on that project alone). Please reopen.

**Comment by pstachula-dev** [comment]
Issue persit, same problem

**Comment by y-hsgw** [comment]
@pstachula-dev 
I believe the issue can be resolved by applying [Aggressive Reporting](https://github.com/testing-library/eslint-plugin-testing-library?tab=readme-ov-file#testing-libraryutils-module) to your ESLint configuration as follows:

```diff
{
  files: ['**/*.test.{ts,tsx}'],
  ...pluginTestingLibrary.configs['flat/react'],
+ settings: {
+   'testing-library/utils-module': 'custom-render',
+ },
},
```

**Comment by pstachula-dev** [comment]
Omg thanks!!!

**Comment by rdamazio** [comment]
> I believe the issue can be resolved by applying [Aggressive Reporting](https://github.com/testing-library/eslint-plugin-testing-library?tab=readme-ov-file#testing-libraryutils-module) to your ESLint configuration as follows:

Thank you - I tried adding that but unfortunately still seeing exactly the same. Am I holding it wrong? https://github.com/rdamazio/efis-editor/compare/bug/no-node-access

**Comment by y-hsgw** [comment]
It seems you're encountering the same issue mentioned in [this comment](https://github.com/testing-library/eslint-plugin-testing-library/issues/1038#issuecomment-3062146982), right?
This appears to be a different problem from the current issue, and it doesn't seem like the `Aggressive Reporting` setting is necessary.
I'm currently looking into whether I can improve this on my end, but it might be better if you create a separate issue.

**Comment by rdamazio** [comment]
> I'm currently looking into whether I can improve this on my end, but it might be better if you create a separate issue.

Thanks for looking into it! Done, #1057 .

**Comment by Belco90** [comment]
> [@pstachula-dev](https://github.com/pstachula-dev) I believe the issue can be resolved by applying [Aggressive Reporting](https://github.com/testing-library/eslint-plugin-testing-library?tab=readme-ov-file&rgh-link-date=2025-08-01T14%3A16%3A46.000Z#testing-libraryutils-module) to your ESLint configuration as follows:
> 
> {
>   files: ['**/*.test.{ts,tsx}'],
>   ...pluginTestingLibrary.configs['flat/react'],
> + settings: {
> +   'testing-library/utils-module': 'custom-render',
> + },
> },

It shouldn't be necessary to set that setting. It fixes the issue because it actually disables the aggressive reporting.

However, the latest issues and fixes we have been doing around reporting things correctly when coming from Testing Library or not, made me think it would be more beneficial to remove the aggressive reporting in the next major. I'm writing all the details in a discussion that I hope I can publish this week so we can keep discussing there.
